src/parser.ts

```diff
@@ -15,13 +15,119 @@ type TokenKind =
   | 'WHITESPACE'
   | 'EOF';
 
+/**
+ * Parses a given string, returning a Tree object.
+ *
+ * @param input
+ * @returns Resulting tree.
+ */
+function parse(input: string) {
```

Comment:
Structurally, I would call this file `index` or `chomTree` and then move the parser functions to a file. Actually after typing it out and really considering it, maybe not. Do what you want.

Reply:
What do you mean? Isn't this file *only* parser functions? Meaning... I've already done that (except that `index` doesn't exist yet).

Reply:
We talked offline about this: there was confusion about the sub functions of parse being a closures. Resolved.